app/src/main/java/com/rockthevote/grommet/ui/eventFlow/CanvasserInfoViewModel.kt

```diff
@@ -0,0 +1,112 @@
+package com.rockthevote.grommet.ui.eventFlow
+
+import android.location.Location
+import androidx.lifecycle.*
+import com.hadilq.liveevent.LiveEvent
+import com.rockthevote.grommet.data.api.model.ApiGeoLocation
+import com.rockthevote.grommet.data.db.dao.SessionDao
+import com.rockthevote.grommet.data.db.model.Session
+import com.rockthevote.grommet.util.coroutines.DispatcherProvider
+import com.rockthevote.grommet.util.coroutines.DispatcherProviderImpl
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.launch
+import pl.charmas.android.reactivelocation.ReactiveLocationProvider
+import timber.log.Timber
+import java.util.*
+
+/**
+ * Created by Mechanical Man on 5/25/20.
+ */
+
+class CanvasserInfoViewModel(
+        private val dispatchers: DispatcherProvider = DispatcherProviderImpl(),
+        private val sessionDao: SessionDao,
+        private val reactiveLocationProvider: ReactiveLocationProvider
+) : ViewModel() {
+
+    val canvasserInfoData = Transformations.map(sessionDao.getSessionWithPartnerInfo()) { result ->
+        result?.let {
+            CanvasserInfoData(
+                    result.partnerInfo?.partnerInfoId ?: 0,
+                    result.partnerInfo?.partnerName ?: "",
+                    result.session?.canvasserName ?: "",
+                    result.session?.openTrackingId ?: "",
+                    result.session?.partnerTrackingId ?: "",
+                    result.session?.deviceId ?: ""
+            )
+        } ?: run {
+            CanvasserInfoData()
+        }
+    }
+
+    private val _effect = LiveEvent<CanvasserInfoState.Effect?>()
+    val effect: LiveData<CanvasserInfoState.Effect?> = _effect
+
+    private val coroutineExceptionHandler = CoroutineExceptionHandler { _, throwable ->
+        // TODO Should we handle this another way? Is it needed?
```

Comment:
I think we can get rid of the TODO around this? Logging is about all we can do in most cases